tests/test_import.py

```diff
@@ -28,6 +28,38 @@ def test_import_json(app, status, warning):
     # :collapsed: work
     assert "collapsed_TEST_01" in html
 
+    # Check absolute path import
+    abs_path_import_html = Path(app.outdir, "subdoc/abs_path_import.html").read_text()
+    assert "small_abs_path_TEST_02" in abs_path_import_html
+
+    # Check relative path import
+    rel_path_import_html = Path(app.outdir, "subdoc/rel_path_import.html").read_text()
+    assert "small_rel_path_TEST_01" in rel_path_import_html
+
+    # Check deprecated relative path import based on conf.py
+    deprec_rel_path_import_html = Path(app.outdir, "subdoc/deprecated_rel_path_import.html").read_text()
+    assert "small_depr_rel_path_TEST_01" in deprec_rel_path_import_html
+
+    warnings = warning.getvalue()
+    assert (
+        "WARNING: Deprecation warning: Relative path must be relative to the current document in future, "
+        "not to the conf.py location. Use a starting '/', like '/needs.json', to make the path relative to conf.py."
+        in warnings
+    )
+
+
+@with_app(buildername="html", srcdir="doc_test/non_exists_file_import")  # , warningiserror=True)
+def test_import_non_exists_json(app, status, warning):
+    # Check non exists file import
+    try:
+        app.build()
+    except ReferenceError as err:
+        assert (
+            err.args[0]
+            == "Could not load needs import file /home/haiyang/work/useblocks/github/sphinxcontrib-needs/tests/"
```

Comment:
``` /home/haiyang/work/useblocks/github/sphinxcontrib-needs/tests/```
This will not work on all machines ;)

Maybe you could check twice.
First for "Could not load needs import file", then for "doc_test/...".

Reply:
I wonder, why this test is green on CI.
Please check, if the this re really working as expected. E.g. by changing the assert-check slightly.

Reply:
Ohh no, seems the check have failed....

Reply:
done.